packages/server/src/tests/utilities/api/base.ts

```diff
@@ -170,7 +189,18 @@ export abstract class TestAPI {
         }
       }
 
-      throw new Error(message)
+      if (response.error) {
+        // Sometimes the error can be between supertest and the app, and when
+        // that happens response.error is sometimes populated with `text` that
+        // gives more detail about the error. The `message` is almost always
+        // useless from what I've seen.
+        if (response.error.text) {
+          response.error.message = response.error.text
+        }
+        throw new Error(message, { cause: response.error })
+      } else {
+        throw new Error(message)
+      }
```

Comment:
Extracting this extra error information is what gave me the clues I needed to find the problems detailed in https://github.com/Budibase/budibase/pull/13359/files#r1539823938.